cmd/kubeadm/app/cmd/upgrade/common_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"bytes"
+	"testing"
+
+	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+)
+
+func TestPrintConfiguration(t *testing.T) {
+	var tests = []struct {
+		cfg           *kubeadmapiext.MasterConfiguration
+		buf           *bytes.Buffer
+		expectedBytes []byte
+	}{
+		{
+			cfg:           nil,
+			expectedBytes: []byte(""),
+		},
+		{
+			cfg: &kubeadmapiext.MasterConfiguration{
+				KubernetesVersion: "v1.7.1",
+			},
+			expectedBytes: []byte(`[upgrade/config] Configuration used:
+	api:
```

Comment:
This seems pretty brittle vs in case you change things, couldn't you just regex parse the output for the key strings that matter?